internal-webapps/org.wso2.bfsi.consent.management.endpoint/src/main/java/org/wso2/bfsi/consent/management/endpoint/api/ConsentValidationEndpoint.java

```diff
@@ -0,0 +1,197 @@
+        while (headersIterator.hasNext()) {
+            String headerName = headersIterator.next();
+            headersMap.put(headerName, requestHeaders.getString(headerName));
+        }
+
+        JSONObject requestPayload = requestData.has(ConsentExtensionConstants.BODY) ?
+                requestData.getJSONObject(ConsentExtensionConstants.BODY) : null;
+        String requestPath = requestData.getString(ConsentExtensionConstants.ELECTED_RESOURCE);
+        String consentId = requestData.getString(ConsentExtensionConstants.CC_CONSENT_ID);
+        String userId = requestData.getString(ConsentExtensionConstants.USER_ID);
+        String clientId = requestData.getString(ConsentExtensionConstants.CLIENT_ID);
+
+        JSONObject resourceParams = requestData.getJSONObject(ConsentExtensionConstants.RESOURCE_PARAMS);
+
+        if (consentId == null) {
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, "Consent Id is mandatory for consent validation");
+        }
+
+        Map<String, String> resourceParamsMap = new HashMap<>();
+        try {
+            //Adding query parameters to the resource map
+            resourceParamsMap = ConsentUtils.addQueryParametersToResourceParamMap(resourceParams);
+        } catch (URISyntaxException e) {
+            log.error("Error while extracting query parameters", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+            "Error while extracting query parameters");
+        }
+
+        ConsentValidateData consentValidateData = new ConsentValidateData(requestHeaders, requestPayload,
+                requestPath, consentId, userId, clientId, resourceParamsMap, headersMap);
+
+        try {
+            DetailedConsentResource consentResource = consentCoreService.getDetailedConsent(consentId);
+            consentValidateData.setComprehensiveConsent(consentResource);
+        } catch (ConsentManagementException e) {
+            log.error("Exception while getting consent", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, "Exception while getting consent");
+        }
+
+        ConsentValidationResult validationResult = new ConsentValidationResult();
+        consentValidator.validate(consentValidateData, validationResult);
+
+        JSONObject information = ConsentUtils.detailedConsentToJSON(
+                consentValidateData.getComprehensiveConsent());
+        information.put(ConsentExtensionConstants.ADDITIONAL_CONSENT_INFO, validationResult.getConsentInformation());
+        validationResult.setConsentInformation(information);
+
+        JSONObject responsePayload;
+        try {
+            responsePayload = validationResult.generatePayload();
+            responsePayload.put(ConsentExtensionConstants.CONSENT_INFO,
+                    ConsentUtils.signJWTWithDefaultKey(validationResult.getConsentInformation().toString()));
+        } catch (Exception e) {
```

Comment:
Shall we catch the specific exception here?

Reply:
the KeyStoreManager used inside this method throws an Exception instead of a specific exception